### ISSUE TITLE
Payload is getting null while reading additional Data.

### ISSUE THREAD

**Issue by AnshulJain1985** [OneSignal/OneSignal-Android-SDK#159]
Payload is getting `null` while reading additional Data.

```Java
@Override
public void notificationOpened(OSNotificationOpenResult result) {
    OSNotificationAction.ActionType actionType = result.action.type;
    JSONObject data = result.notification.payload.additionalData;
}    
```
```Java
Fatal Exception: java.lang.RuntimeException: Unable to start receiver com.onesignal.NotificationOpenedReceiver: java.lang.NullPointerException: Attempt to read from field 'org.json.JSONObject com.onesignal.OSNotificationPayload.additionalData' on a null object reference
       at android.app.ActivityThread.handleReceiver(ActivityThread.java:3642)
       at android.app.ActivityThread.access$2000(ActivityThread.java:222)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1877)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:158)
       at android.app.ActivityThread.main(ActivityThread.java:7229)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
Caused by java.lang.NullPointerException: Attempt to read from field 'org.json.JSONObject com.onesignal.OSNotificationPayload.additionalData' on a null object reference
       at com.mobileappexperts.roadcast.helper.oneSignalNotificationHandler.OneSignalHandler.notificationOpened(OneSignalHandler.java:34)
       at com.onesignal.OneSignal.fireNotificationOpenedHandler(OneSignal.java:1009)
       at com.onesignal.OneSignal.runNotificationOpenedCallback(OneSignal.java:954)
       at com.onesignal.OneSignal.handleNotificationOpen(OneSignal.java:1041)
       at com.onesignal.NotificationOpenedProcessor.processIntent(NotificationOpenedProcessor.java:101)
       at com.onesignal.NotificationOpenedProcessor.processFromActivity(NotificationOpenedProcessor.java:57)
       at com.onesignal.NotificationOpenedReceiver.onReceive(NotificationOpenedReceiver.java:11)
       at android.app.ActivityThread.handleReceiver(ActivityThread.java:3635)
       at android.app.ActivityThread.access$2000(ActivityThread.java:222)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1877)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:158)
       at android.app.ActivityThread.main(ActivityThread.java:7229)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
```

**Comment by jkasten2** [comment]
@AnshulJain1985 Could you include the version of the SDK you are using?

If you created the notification from the REST API please include the full JSON payload used to create it. If not please include all the options you set on the dashboard.

Can you verify line 34 in your `OneSignalHandler.java` is the line below?
`JSONObject data = result.notification.payload.additionalData;`

Thanks.

**Comment by jkasten2** [comment]
@AnshulJain1985 Closing due to inactivity. Please feel free to open this issue if you have more details.

**Comment by gbtt** [comment]
Hi, I am getting the same problem.
OneSignal Version: 3.4.3
I create the notification from dashboard, with the default settings.
The receivedResult.payload.rawPayload is ok, while receivedResult.payload.additionalData is null

Thanks.

**Comment by jkasten2** [comment]
@gbtt Can you attach the full `rawPayload`?

**Comment by gbtt** [comment]
@jkasten2 
{"google.sent_time":1490623408158,"custom":"{\"i\":\"db23aabb-32da-4d3f-839c-f3cf665979dc\"}","from":"86199884594","alert":"mess","title":"title","google.message_id":"0:1490623408171113%9c0f0ef4f9fd7ecd"}

**Comment by jkasten2** [comment]
@gbtt I don't see `additionalData` in the payload. It should show under `custom` with an `a` key like the following.
```JSON
{
"custom": {"a": {"foo": "bar"} }
}
```

I checked the notification in our system and I don't see it set either. Can you double check you are setting the `data` field on the OneSignal create notification REST API call?

**Comment by jkasten2** [comment]
@gbtt Please contact OneSignal support while signed into if you are still seeing an issue and we can check your other notifications.

@AnshulJain1985 If your crash is happening on the line noted below we have just fixed this in commit 777b7b561abdafca237b3c25dcf423017fe57b3d and will be included in the 3.4.4 release.
```Java
JSONObject data = result.notification.payload.additionalData;
```
The issue is related to old summary notifications, can see more details about this in issue #146.
